### PR TITLE
Login Page

### PR DESCRIPTION
**Pull request by lattai**
Added a login page using UserService API. As of now, emails are not stored, and have no interaction with the index page or the comments page. That will be my next step.
See at : https://leahattai-step-2020.appspot.com/login.html

### CODE REVIEW

File: portfolio/src/main/webapp/login.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <script src="login-script.js"></script>
+  </head>
+  <body onload="login()" >
+       <div id= "heading">
+        <div id = "rectangle">
+            <h1>Leah Attai</h1><h3 >Portfolio</h3>
+        </div>
+    </div>
+    <div id = "content">
+
```
Comment (lzoebaker) [inline portfolio/src/main/webapp/login.html:18]: weird space.

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+public class UserAuthServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String username = userService.getCurrentUser().getUsername();
+      String urlToRedirectToAfterUserLogsOut = "/";
```
Comment (lzoebaker) [inline walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java:35]: This string could be changed to a private final constant?

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      response.getWriter().println("<p>Hello " + username + "!</p>");
+      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    } else {
+      String urlToRedirectToAfterUserLogsIn = "/";
```
Comment (lzoebaker) [inline walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java:41]: This string as well.

File: portfolio/src/main/webapp/login-script.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+/**
+ * Fetches the UserService to log in users.
+ */
+function login() {
+    const responsePromise = fetch('/login');
+    responsePromise.then(response => response.text())
+    .then(authInfo => {
+        document.getElementById("content").innerHTML = authInfo;
+    });
+}
```
Comment (menynexi) [inline portfolio/src/main/webapp/login-script.js:24]: new line

File: portfolio/src/main/webapp/login.html (hunk 1 above)
Comment (menynexi) [reply portfolio/src/main/webapp/login.html:18]: +1

File: portfolio/src/main/webapp/login.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Comments - Leah Attai</title>
+    <link rel="stylesheet" href="style.css">
+    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Permanent Marker" >
+    <script src="login-script.js"></script>
+  </head>
+  <body onload="login()" >
+       <div id= "heading">
+        <div id = "rectangle">
```
Comment (menynexi) [inline portfolio/src/main/webapp/login.html:13]: id needs a space

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/")
```
Comment (menynexi) [inline walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java:25]: I don' think "/" is appropriate

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java (hunk 2 above)
Comment (menynexi) [reply walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java:35]: +1

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+public class UserAuthServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
```
Comment (menynexi) [inline walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java:30]: private static final String TEXT_CONTENT_TYPE = "text/html"; you can re use this variable too

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java (hunk 3 above)
Comment (menynexi) [reply walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java:41]: +1

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    } else {
+      String urlToRedirectToAfterUserLogsIn = "/";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+      response.getWriter().println("<p>Hello stranger.</p>");
+      response.getWriter().println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
+    }
+  }
+}
```
Comment (menynexi) [inline walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java:48]: new line

File: walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      response.getWriter().println("<p>Hello " + username + "!</p>");
+      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    } else {
+      String urlToRedirectToAfterUserLogsIn = "/";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+      response.getWriter().println("<p>Hello stranger.</p>");
```
Comment (menynexi) [inline walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/UserAuthServlet.java:44]: I like this it shows some humor

File: portfolio/src/main/java/com/google/sps/servlets/UserAuthServlet.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        response.setContentType(TEXT_CONTENT_TYPE);
+
+        UserService userService = UserServiceFactory.getUserService();
+        if (userService.isUserLoggedIn()) {
+            String userEmail = userService.getCurrentUser().getEmail();
+            String urlToRedirectToAfterUserLogsOut = LOGIN_URL;
+            String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+            response.getWriter().println("<p>Hello " + userEmail + "!</p>");
+            response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
```
Comment (DirtGrubDylan) [inline portfolio/src/main/java/com/google/sps/servlets/UserAuthServlet.java:43]: Maybe make methods that take emails, or urls, and populate the println results

**Pull request merged**